### PR TITLE
cereal 1.2.2 (add a "with-cmake" option to export cereal-config.cmake)

### PR DESCRIPTION
**Pull request by tvatter**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/cereal.rb [hunk 1]
```diff
@@ -29,9 +32,17 @@ def install
         system "cmake", "..", *std_cmake_args
         system "make"
         system "make", "test"
+        system "make", "install"
+      end
+    elsif build.with? "cmake"
```
Comment (dunn) [inline Formula/cereal.rb]: Seems like the `test` and `cmake` branches could be combined; something like
```ruby
if build.with?("test") || build.with?("cmake")
  ENV.cxx11
  args = std_cmake_args
  args << "-DJUST_INSTALL_CEREAL=ON" if build.with? "cmake"
  system "cmake", "..", *args
  system "make", "test" if build.with? "test"
  system "make", "install"

Comment (tvatter) [inline Formula/cereal.rb]: Totally agreed, I just committed your suggestion.

**Comment by ilovezfs** [comment]
Thanks for your first contribution to Homebrew, @tvatter! Without people like you submitting PRs we couldn't run this project. You rock!

**Comment by tvatter** [comment]
My pleasure!

By the way, are you sure that (a) remove the tests and (b) forcing people to have cmake (although simply copying the headers is enough) are good ideas?

**Comment by ilovezfs** [comment]
@tvatter analytics show no one ever used the test option (and we generally do not run make check or make test anymore in homebrew/core unless it's security software), and cmake is only required if you pass --build-from-source. If you just pour the bottle, you get the headers and the cereal-config.cmake but don't actually need to have cmake installed.

**Comment by tvatter** [comment]
OK, makes sense.

**Pull request closed without merging**